### PR TITLE
Implemented Spreadsheet View on Input Page

### PR DESCRIPTION
**Pull request by jackratcliffe**
**Changes**
The current dataset is now loaded onto the input page. It is editable, with changes persisting throughout the project. A save dataset button has been added to save these changes as a new file. Additionally, the load dataset but is now functional.

### CODE REVIEW

**Comment by ScottWalkerAU** [comment]
@lachiesainsbury You can ignore any file in the review from api/, it's an artifact of merging off from feature/newdatastruct (Which is already merged).
Verify you can run the client and cli before approving.

**Comment by jackratcliffe** [comment]
@lachiesainsbury @ScottWalkerAU The only files I worked on were InputDataView.fxml, InputDataController.java, ProjectTreeController.java and DataManager.java. 

The DataManager class is in api/ however, I only personally added the saveDatasetToFile method.

File: client/src/main/java/org/iconic/project/InputDataController.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    private final WorkspaceService workspaceService;
+
+    @FXML
+    private SpreadsheetView spreadsheet;
+
+    @Inject
+    public InputDataController(final ProjectService projectService, final WorkspaceService workspaceService) {
+        this.projectService = projectService;
+        this.workspaceService = workspaceService;
+
```
Comment (jayden626) [inline client/src/main/java/org/iconic/project/InputDataController.java]: whitespace

File: client/src/main/java/org/iconic/project/InputDataController.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                String cellContents = String.valueOf(dataManager.get().getSampleRow(row).get(column));
+                SpreadsheetCell nextCell = SpreadsheetCellType.STRING.createCell(row, column, 1, 1, cellContents);
+                int changedRow = row;
+                int changedColumn = column;
+                nextCell.itemProperty().addListener((observable, oldValue, newValue) -> {
+                    try{
+                        Number newNumber = Double.parseDouble(String.valueOf(newValue));
+                        updateProjectDataset(changedRow,changedColumn,newNumber);
+                    }catch (Exception e) {
+                        //handle exception here
```
Comment (jayden626) [inline client/src/main/java/org/iconic/project/InputDataController.java]: remove comment

**Pull request merged**